doc/release.md

```diff
@@ -7,15 +7,33 @@ So, you want to release the `X.Y.Z` version of astroid ?
 (Consider triggering the "release tests" workflow in GitHub Actions first.)
 
 1. Check if the dependencies of the package are correct
-2. Check the result (Do `git diff vX.Y.Z-1 ChangeLog` in particular).
-3. Install the release dependencies `pip3 install pre-commit tbump`
-4. Bump the version and release by using `tbump X.Y.Z --no-push`.
-5. Push the tag.
-6. Release the version on GitHub with the same name as the tag and copy and paste the
+2. (If you're releasing a minor (`X.Y.0`), remove the empty changelog for the last
+   unreleased patch version `X.Y-1.Z'`.)
+3. Check the result of `git diff vX.Y.Z-1 ChangeLog` or `git diff vX.Y-1.Z' ChangeLog`.
+4. Install the release dependencies `pip3 install pre-commit tbump`
+5. Bump the version and release by using `tbump X.Y.Z --no-push`.
+6. Check the result and push the tag.
+7. Release the version on GitHub with the same name as the tag and copy and paste the
    appropriate changelog in the description. This trigger the pypi release.
+8. (If you just released a minor (`X.Y.0`), also release what was not yet released on
+   the `X.Y-1` branch then delete it.)
 
 ## Post release
 
+### Backport fixes from main
+
+When a crash or a bug is fixed on the main branch, and it needs backport:
+
+- Make sure that the changelog is for the patch version `X.Y-1.Z'`.
+- (Create a `X.Y-1` branch from the `X.Y-1.0` tag if it does not already exist.)
+- After the PR is merged cherry-pick the commits on the `X.Y-1` branch
+- Do a release for `X.Y-1.Z`.
+- Bump `X.Y-1.Z` branch to `X.Y-1.Z+1-dev0`
+- Merge the `X.Y-1` branch on the main branch. The main branch should have the changelog
```

Comment:
We should make the merge strategy explicit here.

Reply:
I'm not absolutely certain that there won't be conflict in the changelog that should be done manually (there's date that could be added on one side and not the other). I think it's better if the releaser think about it in case of conflict.

Reply:
More precisely I think about the new changelog for 2.10.2 added on the fix branch which add text very similar to the changelog for 2.11.0 and there will be new entry for 2.11 feature on the main branch.

Reply:
Yes, but these are (again) issues that stem from whether we merge `2.10.2` into `main` anyway. You could also say that when you release `2.11` the releaser makes a manual commit which copies the `2.10.x` changelogs to `main` and then drops the `2.10` branch without any merging.
If we do want to merge there should at least be something about `git merge --no-edit --strategy=ours` in the documentation.

Reply:
> If we do want to merge there should at least be something about git merge --no-edit --strategy=ours in the documentation.

We're forced to merge so that ``pre-commit autoupdate`` works. The question is whether or not the merge is ``no edit`` or the one where we make sure the changelog modification are coherent.

> Yes, but these are (again) issues that stem from whether we merge 2.10.2 into main anyway. You could also say that when you release 2.11 the releaser makes a manual commit which copies the 2.10.x changelogs to main and then drops the 2.10 branch without any merging.

I think both option work and that we're well past the point of reaching this https://xkcd.com/1445/ 😄